Platform/obc/hal/comms_hal.c

```diff
@@ -139,9 +137,38 @@ STX_return HAL_S_getFS (float * S_firmware_Version) {
   #endif
 }
 
-/*
-void HAL_S_setFreq (uint32_t S_freq){
-#ifdef Stubbed
-    S_freq =
+STX_return HAL_S_setFreq (float S_freq_new) {
```

Comment:
`S_freq_new' is not the value it was when the function is called in `communication_service.c`.

Reply:
Also, I skipped the mutexes for now. We can add them once this problem gets resolved.